src/lib.rs

```diff
@@ -1050,6 +1050,31 @@ impl<T> Drop for Receiver<T> {
                 // SAFETY: see safety comment at top of function
                 unsafe { dealloc(self.channel_ptr) };
             }
+            // The sender has observed the RECEIVING state and is currently reading the waker from
+            // a poll. We need to loop here until we observe the MESSAGE or DISCONNECTED state.
+            // We busy loop here since we know the sender is done very soon.
+            #[cfg(any(feature = "std", feature = "async"))]
+            UNPARKING => loop {
+                hint::spin_loop();
+                // ORDERING: The load above has already synchronized with the write of the message.
```

Comment:
Maybe replace "load above" with "swap above" to make it clearer what operation this is talking about.